src/async_impl/client.rs

```diff
@@ -526,13 +527,16 @@ impl Client {
         let (
             method,
             url,
-            user_headers,
+            mut headers,
             body
         ) = req.pieces();
 
-        let mut headers = self.inner.headers.clone(); // default headers
-        for (key, value) in user_headers.iter() {
-            headers.insert(key, value.clone());
+        let default_headers = self.inner.headers.clone(); // default headers
```

Comment:
We don't need to clone the headers here, since we'll be inserting into the other map, yea?

Reply:
That's true! Fixing.